src/main/java/org/checkstyle/autofix/CheckstyleAutoFix.java

```diff
@@ -49,12 +69,38 @@ public String getViolationReportPath() {
         return violationReportPath;
     }
 
+    public String getCheckstyleConfigurationPath() {
+        return checkstyleConfigurationPath;
+    }
+
+    public String getPropertiesPath() {
+        return propertiesPath;
+    }
+
     @Override
     public List<Recipe> getRecipeList() {
-
         final List<CheckstyleViolation> violations = CheckstyleReportParser
                 .parse(Path.of(getViolationReportPath()));
 
         return CheckstyleRecipeRegistry.getRecipes(violations);
     }
+
+    private Configuration loadCheckstyleConfiguration() throws CheckstyleException, IOException {
+        Properties props = new Properties();
+        final String propFile = getPropertiesPath();
+
+        if (propFile == null) {
+            props = System.getProperties();
+        }
+        else {
+            try (FileInputStream input = new FileInputStream(propFile)) {
+                props.load(input);
+            }
+            catch (FileNotFoundException exception) {
+                throw new IllegalArgumentException("Failed to read: " + propFile, exception);
+            }
+        }
+        return (Configuration) ConfigurationLoader.loadConfiguration(
```

Comment:
you are trying to case internal checkstyle class to your own, I think you will get runtime exception here when you try to run it